src/lib/y2storage/callbacks/initialize.rb

```diff
@@ -0,0 +1,111 @@
+# encoding: utf-8
+
+# Copyright (c) 2018 SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "yast/i18n"
+require "yast2/popup"
+
+Yast.import "Mode"
+Yast.import "Label"
+
+module Y2Storage
+  module Callbacks
+    # Class to implement callbacks used when initializing the storage instance
+    class Initialize
+      include Yast
+      include Yast::Logger
+      include Yast::I18n
+
+      # Constructor
+      #
+      # @param error [Storage::LockException]
+      def initialize(error)
+        textdomain "storage"
+
+        @error = error
+      end
+
+      # Callback to ask the user whether to retry or abort when the storage lock
+      # cannot be acquired.
+      #
+      # @return [Boolean] true if the user decides to retry.
+      def retry?
+        log.info "Storage subsystem is locked by process #{locker_pid}, asking to user whether to retry"
+
+        if locker_name
+          message = format(
+            # TRANSLATORS: %{name} is replaced by the name of a process (e.g., yast2)
+            # and %{pid} by the pid of a process (e.g., 5032).
+            _("The storage subsystem is locked by the application \"%{name}\" (%{pid})."),
+            name: locker_name,
+            pid:  locker_pid
+          )
+        else
+          message = _("The storage subsystem is locked by an unknown application.")
+        end
+
+        message << "\n"
+        message << _("You must quit that application before you can continue.")
+        message << "\n\n"
+        message << _("Would you like to abort or try again?")
+
+        headline = _("Accessing the storage subsystem failed")
+
+        buttons = { yes: _("Retry"), no: abort_button_label }
+
+        answer = Yast2::Popup.show(message, headline: headline, buttons: buttons)
```

Comment:
Shouldn't all callbacks use `Report` instead of directly `Popup`. After all, they can be raised during AutoYaST.